README.md

```diff
@@ -2,16 +2,57 @@
 
 An easy to use client for the [Flagr](https://checkr.github.io/flagr) feature flag service.
 
+## General Concepts used in this documentation
+ * **Flag** can be a feature flag, an experiment, or a configuration.
+ * **Context** is the information that needs to be provided to evaluate a **flag**.
+ * **Variant** represents the possible variation of a flag.
+ * **Variant Attachment** represents dynamic properties that can be added to the **context**. With this it's possible to rules on which **variant** to return.
+
+For more in depth information visit [Flagr official docs](https://checkr.github.io/flagr/#/home).
+
 ## Usage
 
+Assume there's a flag called `color` for changing the blue background color of a page for a red one for 10% of the clients. Here is what the code for evaluating this flag would look like:
+
 ````java
 flagr = new Flagr(FLAGR_HOST);
-context = new EvaluationContext("color_flag");
 try {
     EvaluationResponse response = flagr.evaluate(
-        new EvaluationContext("color_flag")
+        new EvaluationContext("background_color")
     );
-    System.out.println(response.getVariantKey()); // example output: "red"
+    System.out.println(response.getVariantKey()); // The variant will be red for 10% of evaluations
+} catch (FlagrException e) {
+    e.printStackTrace();
+}
+````
+
+If one of the clients is testing the red background and only this client is supposed to see the red background?
+The code above can send the client properties to Flagr so it knows about the client. Then on Flagr Admin UI you can specify that only your test client will see the red background.
+The code for this example would look like:
+````java
+flagr = new Flagr(FLAGR_HOST);
+try {
+    EvaluationContext context = new EvaluationContext("background_color");
+    context.setEntityContext(client, Client.class); // Sends client info to Flagr so you can filter by one of it's properties on the UI.
+    EvaluationResponse response = flagr.evaluate(context);
+    System.out.println(response.getVariantKey()); // This would return red for the client(s) you select on Flagr UI.
+} catch (FlagrException e) {
+    e.printStackTrace();
+}
+````
+
+If the variant needs properties as well, this can be achieved by creating
+a variant attachment and describing it's properties on Flagr UI. Suppose
+color has the properties name and hexadecimal, the code would look like:
+````java
+flagr = new Flagr(FLAGR_HOST);
+try {
+    EvaluationContext context = new EvaluationContext("background_color");
+    context.setEntityContext(client, Client.class); // Sends client info to Flagr so you can filter by one of it's properties on the UI.
+    EvaluationResponse response = flagr.evaluate(context);
+    Color color = response.getVariantAttachment(Color.class);
```

Comment:
This is okay but wouldn't it be better to do
```suggestion
    EvaluationResponse<Color> response = flagr.evaluate(context, Color.class);
    Color color = response.getVariantAttachment();
```

Reply:
The code for that version of the method won't compile because it will need a cast to `EvaluationResponse<Color>` at the end of the evaluation method and java doesn't support casting to parameterized types aparently. :/

Reply:
@blacksails, I'll add some other syntax sugar by having an evaluation that returns the variant.